### PR TITLE
Add use case to Developer Guide

### PR DESCRIPTION
**Pull request by sltsheryl**
Update the Developer Guide with more use cases. 
Textual user cases for creating user, deleting user, add a tag, search a user.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -285,30 +285,132 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-4.  AddressBook deletes the person
+**Extensions**
+* 4a. OBS detects an error in the entered command.
+    * 4a1. OBS requests for the correct command.
+    * 4a2. User enters new command.
+    * Steps 4a1-4a2 are repeated until the command entered are correct. \
+      Use case resumes from step 4.
+
+
+* a. At any time, User chooses to close the AddressBook.
```
[the remark this answers is not in the document]
Comment (sltsheryl) [reply docs/DeveloperGuide.md]: Updated with the below changes. Also changed OBS to MassLinkers. Previously was mistaken.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -283,32 +283,134 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * a1. MassLinkers updates the written storage file.
+    * a2. MassLinkers closes the program.
+      \
+      Use case ends.
+
+**Use case 3: Search classmate via a tag**
+
+**MSS**
+
+1. Student searches students with a particular tag.
```
Comment (carriezhengjr) [inline docs/DeveloperGuide.md:342]: searches classmates? 🤔

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -283,32 +283,134 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2. MassLinkers verifies syntax of the search command is valid.
+3. MassLinkers searches through the list of added tags.
+4. MassLinkers returns the list of students associated with the particular tag and display it through the GUI.
```
Comment (carriezhengjr) [inline docs/DeveloperGuide.md:345]: the list of classmates? 🤔

**Review by carriezhengjr — COMMENTED** [review]
LGTM, just 2 areas where students could be changed to classmates. 😄

**Review by carriezhengjr — APPROVED** [review]
LGTM

**Pull request merged**